src/Shared/DiagnosticDefinitions.cs

```diff
@@ -0,0 +1,27 @@
+// <copyright file="DiagnosticDefinitions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+namespace OpenTelemetry.Internal;
+
+internal static class DiagnosticDefinitions
+{
+    public const string UrlFormat = "https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/diagnostics/{0}.md";
```

Comment:
I don't think we could ever delete one. Because some shipped code out there will still be pointing to it. All we can do is update say the `OTEL1000.md` file to note at the top (something like): "This was released as stable in vX.Y.Z and is no longer marked experimental going forward." Or perhaps: "This feature was completely removed in vX.Y.Z and is no longer available going forward."

Should we have individual files or one big one? 🤷 Seems easier to maintain them individually but not sure about that.

Looks like StyleCop has 3 levels. An index [showing the categories](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/DOCUMENTATION.md). An index for [each category](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SpacingRules.md). And then [individual files](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1000.md). When I get a warning from StyleCop in VS (just tested CA1311) it links me directly to the individual file.

Reply:
I envisioned there would still be individual files, but they could be deleted and links on the index page could be updated to point to a prior permalinked tag/commit. I too noted that codes for other things (stylecop/xunit/csharp), typically link to a dedicated page. Main reason I was considering a different approach is that experimental features are typically short lived. Though no strong opinion here.

Reply:
OK you convinced me this was useful so I implemented it. Take a look at latest and LMK. I also renamed things from "Experimental Feature" to "Experimental API" in the code & docs. I thought that would help differentiate these from the experimental features we typically hide behind envvars.

Reply:
Cool, looks good! Though bear with me through a little more bikeshedding...

I think the name of the folder `diagnostics` may get confused with other things. I suggest an organization like:

```
docs/
└── codeanalysis/
    ├── experimental-apis/
    │   ├── index.md
    │   ├── OTEL1000.md
    │   └── OTEL1001.md
    └── some-future-category/
        └── OTEL2000.md
```